File: FrozenApp/src/main/java/com/github/lakeshire/frozen/MainFragment.java
package com.github.lakeshire.frozen;

import android.content.Intent;
import android.view.View;

import com.xys.libzxing.zxing.activity.CaptureActivity;

import java.util.ArrayList;
import java.util.List;

import lakeshire.github.com.frozenframework.fragment.IPager;
import lakeshire.github.com.frozenframework.fragment.common.AbsBottomBarFragment;

public class MainFragment extends AbsBottomBarFragment {
    @Override
    public void initUi() {
        super.initUi();
        mPager.setOffscreenPageLimit(3);
    }

    @Override
    protected List<IPager> initFragments() {
        List<IPager> fragments = new ArrayList<>();

        for (int i = 0; i < 3; i++) {
            MultiItemFragment fragment = new MultiItemFragment();
            fragment.setTabTitle(getTitles()[i]);
            fragments.add(fragment);
        }

        return fragments;
    }

    @Override
    protected String[] getTitles() {
        return new String[] {"首页", "分类", "我的"};
    }

    @Override
    protected int[] getIcons() {
        return new int[] { R.drawable.ic_home, R.drawable.ic_category, R.drawable.ic_me };
    }

    @Override
    protected int[] getActiveColors() {
        return new int[] { R.color.colorPrimary, R.color.orange, R.color.colorAccent};
    }

    @Override
    protected String getTitle() {
        return "我的应用";
    }

    @Override
    protected int getActionRes() {
        return R.drawable.ic_delete;
    }

    @Override
    protected View.OnClickListener getActionListener() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivityForResult(new Intent(mContext, CaptureActivity.class), 0);
            }
        };
    }

    /**
     * 主Fragment必须重写这个 表示由Activity处理后退事件
     * @return
     */
    @Override
    public boolean onBackPressed() {
        return false;
    }
}

File: FrozenApp/src/main/java/com/github/lakeshire/frozen/MainFragment.kt
package com.github.lakeshire.frozen

import android.view.View
import lakeshire.github.com.frozenframework.fragment.IPager
import lakeshire.github.com.frozenframework.fragment.common.AbsBottomBarFragment
import java.util.*

class MainFragment : AbsBottomBarFragment() {
    override fun initUi() {
        super.initUi()
        mPager.offscreenPageLimit = 3
    }

    override fun initFragments(): List<IPager> {
        val fragments = ArrayList<IPager>()

        for (i in 0..2) {
            val fragment = MultiItemFragment()
            fragment.tabTitle = titles[i]
            fragments.add(fragment)
        }

        return fragments
    }

    override val titles: Array<String>
        get() = arrayOf("首页", "分类", "我的")

    override val icons: IntArray
        get() = intArrayOf(R.drawable.ic_home, R.drawable.ic_category, R.drawable.ic_me)

    override val activeColors: IntArray
        get() = intArrayOf(R.color.colorPrimary, R.color.orange, R.color.colorAccent)

    override fun getTitle(): String {
        return "我的应用"
    }

    override fun getActionRes(): Int {
        return R.drawable.ic_delete
    }

    override fun getActionListener(): View.OnClickListener? {
        return View.OnClickListener {
            val dialog = PlayFragment()
            dialog.show(fragmentManager, "dialog")
        }
    }

    /**
     * 主Fragment必须重写这个 表示由Activity处理后退事件
     * @return
     */
    override fun onBackPressed(): Boolean {
        return false
    }
}
